search-commons/src/main/java/no/unit/nva/search2/model/GatewayResponse.java

```diff
@@ -0,0 +1,41 @@
+package no.unit.nva.search2.model;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Map;
+
+import static no.unit.nva.commons.json.JsonUtils.dtoObjectMapper;
+
+@SuppressWarnings("PMD.ShortMethodName")
+public record GatewayResponse<T>(
+    T body,
+    int statusCode,
+    Map<String, String> headers) {
+
+    public static GatewayResponse<SwsOpenSearchResponse> of(InputStream inputStream) throws IOException {
+        var typeReference = new TypeReference<GatewayResponse<SwsOpenSearchResponse>>() {
+        };
+        return dtoObjectMapper.readValue(inputStream, typeReference);
```

Comment:
should it return this as it is or a `new GatewayResponse` like you have further down in the other `of` method?